script/train_entered.lua

```diff
@@ -34,141 +49,157 @@ local function create_temp_train(event,position)
+	TrainInTunnel.currently_burning = event.cause.burner.currently_burning
+	TrainInTunnel.remaining_burning_fuel = event.cause.burner.remaining_burning_fuel
+	TrainInTunnel.fuel_inventory = event.cause.get_fuel_inventory().get_contents()
+	TrainInTunnel.len_carriages = #event.cause.train.carriages
+	TrainInTunnel.real_carriages = event.cause.train.carriages
 
 	--save carriage information
 	if #event.cause.train.carriages > 1 then
-		global.TrainsInTunnel[SpookyGhost.unit_number].carriages = {}
+		TrainInTunnel.carriages = {}
 		for i=2,#event.cause.train.carriages,1 do
 			event.cause.train.carriages[i].train.speed = 0.5
-			global.TrainsInTunnel[SpookyGhost.unit_number].carriages[i]={}
+			TrainInTunnel.carriages[i]={}
 			if event.cause.train.carriages[i].type == "cargo-wagon" then
-				global.TrainsInTunnel[SpookyGhost.unit_number].carriages[i].type = "cargo-wagon"
-				global.TrainsInTunnel[SpookyGhost.unit_number].carriages[i].cargo = event.cause.train.carriages[i].get_inventory(defines.inventory.cargo_wagon).get_contents()
-				global.TrainsInTunnel[SpookyGhost.unit_number].carriages[i].bar = event.cause.train.carriages[i].get_inventory(defines.inventory.cargo_wagon).get_bar()
-				global.TrainsInTunnel[SpookyGhost.unit_number].carriages[i].filter = {}
+				TrainInTunnel.carriages[i].type = "cargo-wagon"
+				TrainInTunnel.carriages[i].cargo = event.cause.train.carriages[i].get_inventory(defines.inventory.cargo_wagon).get_contents()
+				TrainInTunnel.carriages[i].bar = event.cause.train.carriages[i].get_inventory(defines.inventory.cargo_wagon).get_bar()
+				TrainInTunnel.carriages[i].filter = {}
 				for j = 1, #event.cause.train.carriages[i].get_inventory(defines.inventory.cargo_wagon) do
-					global.TrainsInTunnel[SpookyGhost.unit_number].carriages[i].filter[j] = event.cause.train.carriages[i].get_inventory(defines.inventory.cargo_wagon).get_filter(j)
+					TrainInTunnel.carriages[i].filter[j] = event.cause.train.carriages[i].get_inventory(defines.inventory.cargo_wagon).get_filter(j)
 				end
 			elseif (event.cause.train.carriages[i].type == "fluid-wagon") then
-				global.TrainsInTunnel[SpookyGhost.unit_number].carriages[i].type = "fluid-wagon"
-				global.TrainsInTunnel[SpookyGhost.unit_number].carriages[i].fluids = event.cause.train.carriages[i].get_fluid_contents()
+				TrainInTunnel.carriages[i].type = "fluid-wagon"
+				TrainInTunnel.carriages[i].fluids = event.cause.train.carriages[i].get_fluid_contents()
 			end
 		end
 	end
 end
 
-local function entity_damaged(event)
-	if event.cause and event.entity then
-		object = event.cause.type
-		objectN = event.cause.name
-	else
-		object = nil
-		objectN = nil
+local function get_orientation(Exit,Entrance)
+	x = global.Tunnels[Exit].tunnel.position.x - global.Tunnels[Entrance].tunnel.position.x
+	y = global.Tunnels[Exit].tunnel.position.y - global.Tunnels[Entrance].tunnel.position.y
+	res = (math.atan2(y, x)+(math.pi/2)) / (math.pi*2)
+	if res > 2*math.pi then
```

Comment:
Shouldn't this check be without math.pi since we divided atan2 by math.pi * 2 already?

Reply:
I already checked in the game. That is the correct calculated angle.